lib/Parser/Parser.cpp

```diff
@@ -46,7 +46,12 @@ Optional<PredicateDecl> Parser::parsePredicateDecl() {
             if(parseParameter().unwrapInto(param)) {
                 parameters.push_back(param);
             } else {
-                emitSyntaxError("Expected an additional argument after \",\" in parameter list.");
+                if (parameters.size() == 0) {
+                    emitSyntaxError("Empty parentheses should not be included for predicates with no arguments.")
+                } else {
+                    emitSyntaxError("Expected an additional argument after \",\" in parameter list.");
+                }
```

Comment:
I just realized there was a pre-existing "bug" here where errors from `parseParameter` will cascade and produce a second diagnostic which is less relevant and less helpful than the first one. This is something Allium would be better off avoiding, but this is irrelevant to your PR. 

As some helpful context, see [Clang's diagnostic docs](https://clang.llvm.org/diagnostics.html)'s "Quality of Implementation" section.